pkg/logs/client/http/destination.go

```diff
@@ -376,15 +379,58 @@ func (d *Destination) updateRetryState(err error, isRetrying chan bool) bool {
 }
 
 func httpClientFactory(timeout time.Duration, cfg pkgconfigmodel.Reader) func() *http.Client {
+
+	transport := httputils.CreateHTTPTransport(cfg)
+
+	// Configure transport based on user setting
+	switch cfg.Get("logs_config.transport_type") {
+	case "http1":
+		transport.TLSNextProto = make(map[string]func(authority string, c *tls.Conn) http.RoundTripper)
+		transport.TLSClientConfig = &tls.Config{
+			NextProtos: []string{"http/1.1"},
+		}
+	case "auto":
+		// Use default ALPN auto-negotiation
```

Comment:
Don't we need to call `http2.ConfigureTransport` to allow the existing transport to negotiate up?

Reply:
yep! I removed it without noticing

Reply:
> (I think http2.ConfigureTransport forces http2, but not 100% sure)
It could be worth testing this behavior. The docs are not clear.